frontend/src/libs/components/portal/portal.tsx

```diff
@@ -0,0 +1,39 @@
+import { type ReactNode } from 'react';
+import ReactDOM from 'react-dom';
+
+import { useEffect, useMemo } from '~/libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: ReactNode;
+};
+
+const Portal: React.FC<Properties> = ({ children }: Properties) => {
+  const portalContainer = useMemo(() => {
+    const element = document.createElement('div');
```

Comment:
Why not add an element for the portal directly to the `index.html`?
For example:
```
   <body>
     <div id="root"></div>
     <div id="modal-root"></div>
     <script type="module" src="/src/index.tsx"></script>
   </body>
```